packages/std/src/imports.rs

```diff
@@ -48,14 +48,19 @@ impl ReadonlyStorage for ExternalStorage {
                 If this is causing trouble for you, have a look at https://github.com/confio/cosmwasm/issues/126");
         } else if read < 0 {
             panic!("An unknown error occurred in the read_db call.")
-        } else if read == 0 {
-            return None;
         }
 
-        unsafe { consume_region(value).ok() }.map(|mut d| {
-            d.truncate(read as usize);
-            d
-        })
+        match unsafe { consume_region(value) } {
```

Comment:
What about `consume_region` returning `Option<Vec<u8>>`, with auto-converting `len == 0` to `None`.
It would make all callers be careful about checking that.

Reply:
I don't like the conversion from empty value to None at all and only kept it to avoid changing something unrelated. I think the caller of `ReadonlyStorage.get` should be able to differentiate between a non-existent key and an empty value for an existing key. Depending on the application you can choose to treat those two things as one case or not.

Reply:
Ok